### ISSUE TITLE
Field with *time.Time `sql:"not null"` not work

### ISSUE THREAD

**Issue by douglarek** [jinzhu/gorm#784]
```
package main

 import (
         "time"

         _ "github.com/go-sql-driver/mysql"
         "github.com/jinzhu/gorm"
 )

 type Base struct {
         ID        uint64 `gorm:"primary_key"`
         CreatedAt time.Time
         UpdatedAt time.Time
         DeletedAt *time.Time `sql:"index"`
         CreatorID uint64     `sql:"not null;index"`
 }

 type Atest struct {
         Base
         Tint uint       `sql:"not null"`
         Tat  *time.Time `sql:"not null"`
 }

 var db gorm.DB

 func init() {
         var err error
         db, err = gorm.Open("mysql", "root:@/mysql?charset=utf8&parseTime=True")
         if err != nil {
                 panic(err)
         }
         db.LogMode(true)
 }

 func main() {
         db.CreateTable(&Atest{})
         // at := Atest{Tint: 2}
         // fmt.Println(db.Debug().Create(&at))
 }
```

```
Error 1067: Invalid default value for 'tat'

CREATE TABLE `atests` (`id` bigint unsigned AUTO_INCREMENT,`created_at` timestamp NULL,`updated_at` timestamp NULL,\
`deleted_at` timestamp NULL,`creator_id` bigint unsigned,`tint` int unsigned NOT NULL ,`tat` timestamp NULL NOT NULL  , PRIMARY KEY (`id`))
```

`tat timestamp NULL NOT NULL` is very strange !

**Comment by alecha** [comment]
why are you declaring a pointer of `time.Time` is the value should be not null in the DB?

**Comment by douglarek** [comment]
@alecha If it is a pointer, it will be nil(Go's `zero value`), otherwise it will be a default time value;

and when you declare a pointer , it will help you find bugs.

**Comment by jinzhu** [comment]
Fixed this in v1.0 branch, thank you for your report.
